library/tinkerpop/src/main/java/uk/gov/gchq/gaffer/tinkerpop/GafferPopEdge.java

```diff
@@ -130,11 +130,7 @@ public Vertex inVertex() {
     private static GafferPopVertex checkVertex(final Object vertexId, final GafferPopGraph graph) {
         final GafferPopVertex gafferPopVertex;
         if (vertexId instanceof Vertex) {
-            if (vertexId instanceof GafferPopVertex) {
```

Comment:
This check is unnecessary as if the vertexId was an instance of GafferPopVertex the method below would execute not the current method.

Reply:
It does look slightly different. Beforehand, it would not need to create a new object if the label was id

Reply:
It does look slightly different. Beforehand, it would not need to create a new object if the label was id

Reply:
Although I think the original code is messy, this change technically changes behaviour as a new object is always created, whereas previously sometimes it is just assigned:
https://github.com/gchq/Gaffer/blob/f03f6aea98a52ccbc530099d72301efa4fc428b0/library/tinkerpop/src/main/java/uk/gov/gchq/gaffer/tinkerpop/GafferPopEdge.java#L141-L144
Maybe this can be cleaned up or just returned to how it was before?

Reply:
On second look, I think it is fine as the other function would be called instead